Allows refs to be released after the handle is set as invalid.

madrang opened the pull request:
After SetHandleAsInvalid the remaining refs can be released by calling DangerousRelease.
Test that the invalid handle will not be released.
Test that DangerousRelease will throw ObjectDisposedException when there is no more refs left.

monoadmin commented:
Hello! I'm the build bot for the Mono project. I need approval from a Mono team member to build this pull request. A team member should reply with "approve" to approve a build of this pull request, "whitelist" to whitelist this and all future pull requests from this contributor, or "build" to explicitly request a build, even if one has already been done. Contributors can ignore this message.

madrang commented:
Just updated the PR.
I added a missing sf.DangerousRelease (); that i missed since the test already fails. But i ran that code under Ms.Net this morning and saw that i was missing one to replicate the Ms.Net behavior.
